src/contracts/account/ProxyAccountDeployer.sol

```diff
@@ -22,63 +36,83 @@ contract ProxyAccount is ReplayProtection {
+     * We check the signature has authorised the call before executing it.
+     * @param _metaTx A single meta-transaction that includes target, value and calldata
+     * @param _replayProtection Replay protection
+     * @param _replayProtectionAuthority Address of external replay protection
      * @param _signature Signature from signer
      */
     function forward(
-        address _target,
-        uint _value, 
-        bytes memory _callData,
+        MetaTx memory _metaTx,
         bytes memory _replayProtection,
         address _replayProtectionAuthority,
         bytes memory _signature) public {
 
         // Assumes that ProxyAccountDeployer is ReplayProtection. 
-        bytes memory encodedData = abi.encode(_target, _value, _callData);
+        bytes memory encodedData = abi.encode(CallType.CALL, _metaTx.target, _metaTx.value, _metaTx.callData);
 
         // // Reverts if fails.
         require(owner == verify(encodedData, _replayProtection, _replayProtectionAuthority, _signature), "Owner did not sign this meta-transaction.");
+        forwardCall(_metaTx.target, _metaTx.value, _metaTx.callData); // We do not want this to revert. Save the replay protection 
+    }
 
-        (bool success, bytes memory revertReason) = _target.call.value(_value)(abi.encodePacked(_callData));
+    /**
+     * We check the signature has authorised the call before executing it.
+     * WARNING: Be VERY VERY VERY cautious of contracts that can modify / store state. 
+     * Intended use is when msg.sender is required for code execution (e.g. CREATE2). 
+     * @param _metaTx A single meta-transaction that includes target, value and calldata
+     * @param _replayProtection Replay protection
+     * @param _replayProtectionAuthority Address of external replay protection
+     * @param _signature Signature from signer
+     */
+    function delegate(
+        MetaTx memory _metaTx,
+        bytes memory _replayProtection,
+        address _replayProtectionAuthority,
+        bytes memory _signature) public {
 
-        if(!success) {
-            assembly {revertReason := add(revertReason, 68)}
-            // 4 bytes = sighash
-            // 64 bytes = length of string
-            // If we slice offchain, then we can verify the sighash
-            // too. https://twitter.com/ricmoo/status/1262156359853920259
-            // IF we slice onchain, then we lose that information.
-            emit Revert(string(revertReason));
-        }
+        // Assumes that ProxyAccountDeployer is ReplayProtection. 
+        bytes memory encodedData = abi.encode(CallType.DELEGATE, _metaTx.target, _metaTx.value, _metaTx.callData);
+
+        // // Reverts if fails.
+        require(owner == verify(encodedData, _replayProtection, _replayProtectionAuthority, _signature), "Owner did not sign this meta-transaction.");
+        delegate(_metaTx.target, _metaTx.callData); // We do not want this to revert. Save the replay protection 
```

Comment:
I think we should add events to all these functions, and they should include the replay protection.

Reply:
I added it in the ReplayProtection.sol. I've named the events Bitflip and Muiltinonce. We can decide if we want to combine into a single event.

Reply:
I would say we want to add it for all replay protections, including external ones, would make it much easier for people to keep nonces up to date. I think a single ReplayProtection(address, bytes) event may be the easiest for consistency.

Reply:
Ive added in ReplayProtection().

Reply:
its called ReplayProtectionInfo()